lib/safer_rails_console/patches/sandbox/transaction_read_only.rb

```diff
@@ -0,0 +1,26 @@
+module SaferRailsConsole
+  module Patches
+    module Sandbox
+      module TransactionReadOnly
+        module ActiveRecord
+          module ConnectionAdapters
+            module PostgreSQLAdapter
+              def begin_db_transaction
+                super
+                execute 'SET TRANSACTION READ ONLY'
```

Comment:
What happens if someone tries to select for update in this mode?

Reply:
`PG::ReadOnlySqlTransaction: ERROR:  cannot execute SELECT FOR UPDATE in a read-only transaction`

Reply:
Would that behavior break certain code-paths?

Reply:
Cool just wanted to ensure this covers locking in read statements also.  I'm not aware of anything that would break. 

Do you typically use this locally to manually verify your changes?  If you run a couple commands that way and everything behaves properly I think that's enough code-path validation.

Reply:
Yep, I manually verify locally for in-house Salsify apps.

I could build out the feature tests here and specs, but it'd be extremely difficult to capture all the various use-cases that this should be catching.  Would be a nice-to-have though.